src/backend/nats.cr

```diff
@@ -0,0 +1,64 @@
+require "nats"
+
+module Cable
+  class NATSBackend < BackendCore
+    register "nats"
+
+    getter nats : NATS::Client do
+      NATS::Client.new(URI.parse(Cable.settings.url))
+    end
+    getter streams = Hash(String, Set(NATS::Subscription)).new { |streams, channel|
+      streams[channel] = Set(NATS::Subscription).new
+    }
```

Comment:
Had to use `{}` instead of `do`/`end` here due to precedence.

The reason we had to include this is because you can't unsubscribe from a NATS subject directly. You have to unsubscribe from the subscription id since you can subscribe to the same subject multiple times.